plugins/package-managers/bazel/src/funTest/assets/projects/synthetic/bazel-expected-output-conan-dependencies.yml

```diff
@@ -0,0 +1,171 @@
+---
+project:
+  id: "Bazel::plugins/package-managers/bazel/src/funTest/assets/projects/synthetic/bazel-conan-dependencies/MODULE.bazel:"
+  definition_file_path: "<REPLACE_DEFINITION_FILE_PATH>"
+  declared_licenses: []
+  declared_licenses_processed: {}
+  vcs:
+    type: "Git"
+    url: "<REPLACE_URL_PROCESSED>"
+    revision: "<REPLACE_REVISION>"
+    path: "<REPLACE_PATH>"
+  vcs_processed:
+    type: "Git"
+    url: "<REPLACE_URL_PROCESSED>"
+    revision: "<REPLACE_REVISION>"
+    path: "<REPLACE_PATH>"
+  homepage_url: ""
+  scopes:
+  - name: "conan_requires"
+    dependencies:
+    - id: "Conan::conanfile.txt:"
+      dependencies:
+      - id: "Conan::fmt:10.1.1"
+  - name: "conan_test_requires"
+    dependencies:
+    - id: "Conan::conanfile.txt:"
+      dependencies:
+      - id: "Conan::gtest:1.16.0"
+  - name: "dev"
+    dependencies: []
+  - name: "main"
+    dependencies:
+    - id: "Bazel::glog:0.5.0"
+      linkage: "STATIC"
+      dependencies:
+      - id: "Bazel::gflags:2.2.2"
+        linkage: "STATIC"
+    - id: "Conan::conanfile.txt:"
+packages:
+- id: "Bazel::gflags:2.2.2"
+  purl: "pkg:bazel/gflags@2.2.2"
+  declared_licenses: []
+  declared_licenses_processed: {}
+  description: ""
+  homepage_url: "https://gflags.github.io/gflags/"
+  binary_artifact:
+    url: ""
+    hash:
+      value: ""
+      algorithm: ""
+  source_artifact:
+    url: "https://github.com/gflags/gflags/archive/refs/tags/v2.2.2.tar.gz"
+    hash:
+      value: "34af2f15cf7367513b352bdcd2493ab14ce43692d2dcd9dfc499492966c64dcf"
+      algorithm: "SHA-256"
+  vcs:
+    type: "Git"
+    url: "https://github.com/gflags/gflags"
+    revision: ""
+    path: ""
+  vcs_processed:
+    type: "Git"
+    url: "https://github.com/gflags/gflags.git"
+    revision: ""
+    path: ""
+- id: "Bazel::glog:0.5.0"
+  purl: "pkg:bazel/glog@0.5.0"
+  declared_licenses: []
+  declared_licenses_processed: {}
+  description: ""
+  homepage_url: "https://github.com/google/glog"
+  binary_artifact:
+    url: ""
+    hash:
+      value: ""
+      algorithm: ""
+  source_artifact:
+    url: "https://github.com/google/glog/archive/refs/tags/v0.5.0.tar.gz"
+    hash:
+      value: "eede71f28371bf39aa69b45de23b329d37214016e2055269b3b5e7cfd40b59f5"
+      algorithm: "SHA-256"
+  vcs:
+    type: "Git"
+    url: "https://github.com/google/glog"
+    revision: ""
+    path: ""
+  vcs_processed:
+    type: "Git"
+    url: "https://github.com/google/glog.git"
+    revision: ""
+    path: ""
+- id: "Conan::conanfile.txt:"
+  purl: "pkg:conan/conanfile.txt"
```

Comment:
Please add class docs to `Bazel` which explain how the Conan integration works, similar to what you write in the commit message. One thing I am missing is an explanation for why this is a package instead of a project dependency, when it actually refers to a definition file in the same repository. I guess it is done like this because Bazel filters the Conan dependencies, but this needs to be explained at least.
Also, the VCS info below should not be empty.

Reply:
I am trying to add the `VcsInfo` but the `ProjectAnalyzerResults` returned by `Conan.resolveDependencies()` only contains `vcs_processed` , not `vcs` :


```
- id: "Conan::conanfile.txt:"
  purl: "pkg:conan/conanfile.txt"
  [...]
  vcs:
    type: ""
    url: ""
    revision: ""
    path: ""
  vcs_processed:
    type: "Git"
    url: "ssh://git@github.com/oss-review-toolkit/ort.git"
    revision: "0989acd6adcb5d63700a03c04ac1b42f06358b57"
    path: "plugins/package-managers/bazel/src/funTest/assets/projects/synthetic/bazel-conan-dependencies"
```

Any idea why ?
Is there maybe some magic function of ORT that should populate it ?

Reply:
I'm not so familiar with the Conan implementation, but you use a `conanfile.txt` in your test and it seems that only `conanfile.py` allows to define VCS metadata, in this case it is expected that only `vcsProcessed` is populated.

Reply:
Then it's ready to review.